CRM/Core/Permission/Drupal8.php

```diff
@@ -48,6 +48,10 @@ class CRM_Core_Permission_Drupal8 extends CRM_Core_Permission_DrupalBase {
    * @return bool
    */
   public function check($str, $userId = NULL) {
+    if (!\Drupal::hasContainer()) {
+      return FALSE;
+    }
```

Comment:
or 
```php
if (!class_exists('Drupal')) {
      return NULL;
    }
```
@dsnopek ?

as I recall in past I hit with an error as ``` unknown class Drupal``` when a core function is invoked before bootstrap.

Reply:
In the error I was hitting, the `\Drupal` class existed, it just hadn't had `\Drupal::setContainer()` called on it yet. So, the autoloading was setup, but the bootstrap not completed yet.